tests/test_1685-IndexedArray-project-parameters.py

```diff
@@ -0,0 +1,15 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import numpy as np
+import pytest  # noqa: F401
+
+import awkward as ak
+
+
+def test():
+    layout = ak.contents.IndexedArray(
+        ak.index.Index64(np.array([0, 1, 2, 3], np.int64)),
+        ak.contents.NumpyArray(np.arange(10), parameters={"this": "that"}),
```

Comment:
Oh, and the outer one should override whatever the inner had. (So the test output should still be `"some" → "other"`, not `"hidden"`.

```suggestion
        ak.contents.NumpyArray(np.arange(10), parameters={"this": "that", "some": "hidden"}),
```